### ISSUE TITLE
what should I pay attention to when implmenting poll?

### ISSUE THREAD

**Issue by shaitao** [tokio-rs/doc-blitz#58]
It's confusing to implement poll,  when and where to use NotReady

**Comment by carllerche** [comment]
Thanks for the feedback.

Do you have specific examples where you got tripped up?

**Comment by jonhoo** [comment]
I completely agree that a write-up of "how you likely want to implement and think about `poll`" is likely a good addition.

**Comment by carllerche** [comment]
@jonhoo What would you say should go in that section?

**Comment by shaitao** [comment]
> 
> 
> Thanks for the feedback.
> 
> Do you have specific examples where you got tripped up? What were you trying to implement? What did your first attempt look like?

It looks like that if the poll of a source future block and all the futures will block.
if I implement
fn poll(&self)->Poll(T,E){
    let x = loading()? //this is not a future
    Ok(Async::Ready(x))  //how to return NotReady
}
